### PR TITLE
soundd/micd: more robust setup procedure

### PR DESCRIPTION
**Pull request by jnewb1**
using alsa and agnos9 we have had a couple crashes due to random setup errors, which I can replicate after ~25 setups on device

examples:

on device

This seems to caused by alsa not being fully setup by the time soundd/micd starts, which causes the query_devices to fail and therefore not find the device. It occurs more often on release builds. I think the retry is the best way around this
```
Traceback (most recent call last):
  File "/data/openpilot/openpilot/selfdrive/manager/process.py", line 40, in launcher
    mod.main()
  File "/data/openpilot/selfdrive/ui/soundd.py", line 152, in main
    s.soundd_thread()
  File "/data/openpilot/selfdrive/ui/soundd.py", line 135, in soundd_thread
    with sd.OutputStream(device=device, channels=1, samplerate=SAMPLE_RATE, callback=self.callback) as stream:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/pyenv/versions/3.11.4/lib/python3.11/site-packages/sounddevice.py", line 1494, in __init__
    _StreamBase.__init__(self, kind='output', wrap_callback='array',
  File "/usr/local/pyenv/versions/3.11.4/lib/python3.11/site-packages/sounddevice.py", line 817, in __init__
    _get_stream_parameters(kind, device, channels, dtype, latency,
  File "/usr/local/pyenv/versions/3.11.4/lib/python3.11/site-packages/sounddevice.py", line 2659, in _get_stream_parameters
    device = _get_device_id(device, kind, raise_on_error=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/pyenv/versions/3.11.4/lib/python3.11/site-packages/sounddevice.py", line 2799, in _get_device_id
    raise ValueError(
ValueError: No output device matching 'sdm845-tavil-snd-card: - (hw:0,0)'
```

in CI
https://github.com/commaai/openpilot/issues/30634

I believe this issue is caused when both soundd and micd attempt to access the same device at the same time. It seems to be fixed by just using the default device on both soundd and micd which seems to better manage this race condition
```
Traceback (most recent call last):
  File "/data/pythonpath/openpilot/selfdrive/manager/process.py", line 40, in launcher
    mod.main()
  File "/data/pythonpath/system/micd.py", line 92, in main
    mic.micd_thread()
  File "/data/pythonpath/system/micd.py", line 84, in micd_thread
    with sd.InputStream(channels=1, samplerate=SAMPLE_RATE, callback=self.callback) as stream:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/pyenv/versions/3.11.4/lib/python3.11/site-packages/sounddevice.py", line 1421, in __init__
    _StreamBase.__init__(self, kind='input', wrap_callback='array',
  File "/usr/local/pyenv/versions/3.11.4/lib/python3.11/site-packages/sounddevice.py", line 898, in __init__
    _check(_lib.Pa_OpenStream(self._ptr, iparameters, oparameters,
  File "/usr/local/pyenv/versions/3.11.4/lib/python3.11/site-packages/sounddevice.py", line 2745, in _check
    raise PortAudioError(errormsg, err, hosterror_info)
sounddevice.PortAudioError: Error opening InputStream: Unanticipated host error [PaErrorCode -9999]: 'Operation already in progress' [ALSA error -114]
```

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Can we fix these without the retry? Retry should be a very last resort for buggy systems that we don't control (e.g. the Quectel GNSS stack).

**Comment by jnewb1** [comment]
set clock at :26
```Dec 08 12:59:26 comma-3894fcc python[4656]: Fri 08 Dec 2023 08:59:26 PM UTC```

alsa started at :41
```Dec 08 12:59:41 comma-3894fcc alsactl[38414]: Hardware is initialized using a generic method```

so alsa takes 15 seconds to start even after the launch script starts

**Comment by adeebshihadeh** [comment]
Can we add a wait for that case instead?

File: system/micd.py [hunk 1]
```diff
@@ -3,14 +3,25 @@
...
+from openpilot.system.hardware import TICI
 
 RATE = 10
 FFT_SAMPLES = 4096
 REFERENCE_SPL = 2e-5  # newtons/m^2
 SAMPLE_RATE = 44100
 
 
+@retry(attempts=7, delay=3)
+def wait_for_devices(sd):
```
Comment (jnewb1) [inline system/micd.py:17]: waiting for sounddevice to query more than 1 device seems to also do the trick, but you still have to force reload portaudio

**Comment by jnewb1** [comment]
```
Traceback (most recent call last):
  File "/data/openpilot/openpilot/selfdrive/manager/process.py", line 40, in launcher
    mod.main()
  File "/data/openpilot/selfdrive/ui/soundd.py", line 152, in main
    s.soundd_thread()
  File "/data/openpilot/selfdrive/ui/soundd.py", line 135, in soundd_thread
    with sd.OutputStream(device=device, channels=1, samplerate=SAMPLE_RATE, callback=self.callback) as stream:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/pyenv/versions/3.11.4/lib/python3.11/site-packages/sounddevice.py", line 1494, in __init__
    _StreamBase.__init__(self, kind='output', wrap_callback='array',
  File "/usr/local/pyenv/versions/3.11.4/lib/python3.11/site-packages/sounddevice.py", line 817, in __init__
    _get_stream_parameters(kind, device, channels, dtype, latency,
  File "/usr/local/pyenv/versions/3.11.4/lib/python3.11/site-packages/sounddevice.py", line 2659, in _get_stream_parameters
    device = _get_device_id(device, kind, raise_on_error=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/pyenv/versions/3.11.4/lib/python3.11/site-packages/sounddevice.py", line 2799, in _get_device_id
    raise ValueError(
ValueError: No output device matching 'sdm845-tavil-snd-card: - (hw:0,0)'
```

reproed almost every time when on a release build or by disabling build.py, and has been fixed by waiting for >1 device

**Pull request merged**